### ISSUE TITLE
Rule `react-a11y-input-elements` incorrectly requiring placeholder for radio, checkbox, and file inputs

### ISSUE THREAD

**Issue by narthollis** [Microsoft/tslint-microsoft-contrib#749]
### Bug Report

-   **`tslint-microsoft-contrib` version**: 6.0.0
-   **TSLint version**: 5.12.0
-   **TypeScript version**: 3.2.2
-   **Running TSLint via**: CLI

#### TypeScript code being linted

```tsx
<input
    type="checkbox"
    checked={props.isSelected}
    onChange={onChange}
/>
```

with `tslint.json` configuration:

```json

  "extends": [
    "tslint:latest",
    "tslint-microsoft-contrib/recommended_ruleset"
  ],

  "rulesDirectory": [
    "tslint-microsoft-contrib"
  ],

  "rules": {
    "quotemark": [true, "single", "jsx-double"],
  }
}
```

#### Actual behavior

ERROR: 40:17    react-a11y-input-elements  Input elements must include default, place-holding characters if empty

#### Expected behavior

Inputs with type checkbox, radio, and, file should not require a placeholder value.

**Comment by JoshuaKGoldberg** [comment]
Agreed, thanks for pointing this out @narthollis! Accepting PRs to limit the 

For future reference: it looks like the following if statement should also check the `type` attribute to make sure it's not one of the above types that don't need a placeholder value.

https://github.com/Microsoft/tslint-microsoft-contrib/blob/6c6e1d1fe6991ae3bb3f267e6f8c468316a00131/src/reactA11yInputElementsRule.ts#L53
